example/main.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"context"
+
+	"github.com/carousell/go-logging"
+	"github.com/carousell/go-logging/gokit"
+)
+
+func main() {
+
+	ctx := context.Background()
+
+	logging.RegisterLogger(gokit.NewLogger())
+
+	ctx = logging.WithLogger(ctx,
+		logging.WithAttr("main1", "main1"),
+		logging.WithAttr("main2", 123),
+	)
+
+	logging.InfoV2(ctx, "before calling foo()")
+	//{"@timestamp":"2023-04-27T00:39:45.573743+08:00","caller":"go-logging/new.go:33","level":"info","main1":"main1","main2":123,"msg":"before calling foo()"}
+
+	foo(ctx)
+	logging.InfoV2(ctx, "after called foo()")
+	//{"@timestamp":"2023-04-27T00:39:45.574079+08:00","caller":"go-logging/new.go:33","level":"info","main1":"main1","main2":123,"msg":"after called foo()"}
+
+	logging.InfoV3(ctx, "msg v3 without attributes", nil)
+	//{"@timestamp":"2023-04-27T00:01:39.701337+05:30","caller":"go-logging/new.go:35","level":"info","main1":"main1","main2":123,"msg":"msg v3 without attributes"}
+
+	logging.InfoV3(ctx, "msg v3", logging.AttrMap{"listing_id": "1000323749", "is_exp_enabled": true})
```

Comment:
Thanks @bhadreswar-ghuku. I was considering taking `map[string]interface{}`. But this looks even better.